eclipse/che-theia
# Add schema for che tasks

### What does this PR do?
Add schema for `che` tasks:
- `type`, `command` and `label` are required properties 
- `target` and `previewUrl` are optional fields
-  `target` contains optional `workingDir` and `component` properties
- additional properties are allowed

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/15179
https://github.com/eclipse/che/issues/13984

### How to test
Unfortunately, content assistant for `tasks.json` is broken at the moment: https://github.com/eclipse-theia/theia/issues/6542, so we can not fully test it. But `che` tasks should be available for running from `Terminal => Run Task` menu.

1. Use the following devfile, which contains the reference for image with changes related to the current PR (you can check it using `Help=> About` dialog).
 
<details>
<summary>Devfile</summary>
 
```
 apiVersion: 1.0.1-beta
metadata:
 name: wksp-test-tasks
projects:
  - name: che-theia
    source:
      type: git
      location: 'https://github.com/eclipse/che-theia.git'
  - name: theia
    source:
      type: git
      location: 'https://github.com/theia-ide/theia.git'
components:
  - 
    alias: che-dev
    type: dockerimage
    image: eclipse/che-theia-dev:next
    mountSources: true
    endpoints:
      - name: "theia-dev"
        port: 3130
        attributes:
          protocol: tcp
          public: 'true'
    memoryLimit: 2Gi

  - 
    alias: theia-editor
    reference: >-
      https://raw.githubusercontent.com/RomanNikitenko/che-plugin-registry/master/v3/plugins/eclipse/che-theia/next/meta.yaml
    type: cheEditor
   
commands:
- name: test CHE task
  actions:
  - type: exec
    component: theia-editor
    command: >
              sleep 2 && echo CHE task is completed
    workdir: /projects/theia

- name: theia:build
  actions:
  - type: exec
    component: che-dev
    command: >
              yarn
    workdir: /projects/theia

- name: che-theia:build
  actions:
  - type: exec
    component: che-dev
    command: >
              yarn
    workdir: /projects/che-theia

```
</details>
 
2. Go to `Terminal => Run Task` 
you can see that all valid `che` tasks are available for running
3. `type`, `command` and `label` are required properties for `che` tasks, so try to remove/break one of these fields using `task.json` file - after that the task is not available for running from `Terminal => Run Task` menu.
Also you can see the warning:
 
![task_warning](https://user-images.githubusercontent.com/5676062/69136694-f85d8380-0ac3-11ea-879d-63a1f0834a75.png)

 4. After fixing broken task configuration the task should be available for running again.

Signed-off-by: Roman Nikitenko <rnikiten@redhat.com>
